web/modules/custom/joinup_event/joinup_event.module

```diff
@@ -22,3 +22,24 @@ function joinup_event_entity_bundle_field_info_alter(array &$fields, EntityTypeI
     }
   }
 }
+
+/**
+ * Implements hook_entity_type_alter().
+ */
+function joinup_event_entity_type_alter(array &$entity_types) {
+  /* @var $entity_types \Drupal\Core\Entity\EntityTypeInterface */
+  if (isset($entity_types['node'])) {
+    $entity_types['node']->addConstraint('OptionalRequiredLocationFields', ['fieldName1' => 'field_location', 'fieldName2' => 'field_event_online_location']);
+  }
+}
+
+/**
+ * Implements template_preprocess_form_element_label().
+ */
+function joinup_event_preprocess_form_element_label(&$variables) {
+  if (isset($variables['element']['#id'])) {
+    if ($variables['element']['#id'] == 'edit-field-location-0-value') {
```

Comment:
The ticket descriptions states: 

> The Event location could have a red asterisk next to it, without a validation

So the asterisk should be on field group label ("Event Location"), not on physical location.

Reply:
I've removed this because we need to use template_preprocess_fieldset() to change its label.

Reply:
It works great now. The red asterisk is on the "Event location" fieldset.